flake.nix

```
@@ -50,6 +51,19 @@
       }
     );
 
+    # Provides checks to invoke with 'nix flake check'
+    checks = forAllSystems (
+      system: let
+        mkCheckArgs = testDirectory: {
+          inherit self;
+          inherit (nixpkgs) lib;
+          inherit testDirectory;
+          pkgs = nixpkgs.legacyPackages.${system};
+        };
+      in
+        import ./modules/tests (mkCheckArgs ./modules/tests/programs)
+    );
```

`mkCheckArgs` is a needless abstraction that really just makes this kind of confusing. Definitely should not be used. We can just supply the calls at import.

```suggestion
    checks = forAllSystems (
       system: 
          import ./modules/tests {
             inherit self;
             inherit (nixpkgs) lib;
             pkgs = nixpkgs.legacyPackages.${system};
             testDirectory = ./modules/tests/programs;
           }
       );
```